packages/block-library/src/gallery/test/helpers.native.js

```diff
@@ -0,0 +1,293 @@
+	MEDIA_UPLOAD_STATE_FAILED,
+	MEDIA_UPLOAD_STATE_RESET,
+} from '@wordpress/block-editor';
+
+/**
+ * Adds a Gallery block via the block picker.
+ *
+ * @return {import('@testing-library/react-native').RenderAPI} A Testing Library screen.
+ */
+export const addGalleryBlock = async () => {
+	const screen = await initializeEditor();
+	const { getByA11yLabel, getByTestId, getByText } = screen;
+
+	fireEvent.press( getByA11yLabel( 'Add block' ) );
+
+	const blockList = getByTestId( 'InserterUI-Blocks' );
+	// onScroll event used to force the FlatList to render all items
+	fireEvent.scroll( blockList, {
+		nativeEvent: {
+			contentOffset: { y: 0, x: 0 },
+			contentSize: { width: 100, height: 100 },
+			layoutMeasurement: { width: 100, height: 100 },
+		},
+	} );
+
+	fireEvent.press( await waitFor( () => getByText( 'Gallery' ) ) );
+
+	return screen;
+};
+
+/**
+ * The gallery items are rendered via the FlatList of the inner block list.
+ * In order to render the items of a FlatList, it's required to trigger the
+ * "onLayout" event. Additionally, the call is wrapped over "executeStoreResolvers"
+ * because the gallery items request the media data associated with the image to
+ * be rendered via the "getMedia" selector.
+ *
+ * @param {import('react-test-renderer').ReactTestInstance} galleryBlock    Gallery block instance to trigger layout event.
+ * @param {Object}                                          [options]       Configuration options for the event.
+ * @param {number}                                          [options.width] Width value to be passed to the event.
+ */
+export const triggerGalleryLayout = async (
+	galleryBlock,
+	{ width = 320 } = {}
+) =>
+	executeStoreResolvers( () =>
+		fireEvent(
+			within( galleryBlock ).getByTestId( 'block-list-wrapper' ),
+			'layout',
+			{
+				nativeEvent: {
+					layout: {
+						width,
+					},
+				},
+			}
+		)
+	);
+
+/**
+ * Initialize the editor with HTML generated of Gallery block.
+ *
+ * @param {Object}  [options]               Configuration options for the initialization.
+ * @param {string}  [options.html]          String of block editor HTML to parse and render.
+ * @param {number}  [options.numberOfItems] Number of gallery items to generate or already included in the provided block editor HTML.
+ * @param {Object}  [options.media]         Contains media data to be used in the generation.
+ * @param {number}  [options.width]         Width to be passed when triggering the "onLayout" event on the Gallery block.
+ * @param {boolean} [options.selected]      Specifies if the Gallery block included in the initial HTML should be automatically selected.
+ * @param {boolean} [options.useLocalUrl]   Specifies if the items should use the local URL instead of the server URL.
+ *
+ * @return {import('@testing-library/react-native').RenderAPI} The Testing Library screen plus the Gallery block React Test instance.
+ */
+export const initializeWithGalleryBlock = async ( {
```

Comment:
I think we should only use `executeStoreResolvers` for actions that we know would trigger store resolvers. The `initializeEditor` call already covers the ones triggered during the editor initialization, so we won't need `executeStoreResolvers` for that.

However, we need it when rendering the gallery items, as they request media information via the [`getMedia` selector](https://github.com/WordPress/gutenberg/blob/2df751aa2ad079b9c015bc159720d61b48780e0c/packages/block-library/src/image/edit.native.js#L861) which in turn triggers store resolvers ([reference](https://github.com/WordPress/gutenberg/blob/2df751aa2ad079b9c015bc159720d61b48780e0c/packages/core-data/src/entities.js#L46-L53)). In any case, this is already covered via the `triggerGalleryLayout` function hence, I don't think we'd need to use `executeStoreResolvers` here too. Additionally, I'd like to note that this function (`triggerGalleryLayout`) is also used in several test cases, so we would need it in any case.

Reply:
I know the integration test helpers, especially `executeStoreResolvers` are a bit tricky, so please let me know if you'd like me to expand any information 🙇 .

Reply:
I didn't feel strongly about this change. I just noticed that it looked considerably different when I compared it to other use cases. I was trying to see if it could be simplified a bit. Thanks for further explaining. I appreciate it.